### PR TITLE
Add GitHub actions to run vitest

### LINKED ISSUE #89: Add pinia testing to GitHub CI testing workflow
# Requirements

- The GitHub workflow should include pinia testing

---

NOTE: This ticket depends on #88

### PR DESCRIPTION
**Pull request by RageZBla**
Resolves #89

*To be merged after #100*

# What changed

- Add new GitHub actions to run vitest, these test are used to test out the state management part of the application

# Testing instructions

Check GitHub actions results.

### CODE REVIEW

**Comment by theyokohamalife** [comment]
@RageZBla It looks like the vitest / cypress runs are failing in the CI

**Comment by RageZBla** [comment]
@theyokohamalife it's expected. Let know if you want to decide the order of merges.

File: .github/workflows/vitest.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: 'Tests: vitest'
```
Comment (ermish) [inline .github/workflows/vitest.yml:1]: File name recommendation: tests-vitest.yml to match for consistency

File: .github/workflows/vitest.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  push:
+    branches: [ "main", "ragezbla/89-vitest-gh-actions" ] # TODO remove after checking it works
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '45 15 * * 4'
+
+jobs:
```
Comment (ermish) [inline .github/workflows/vitest.yml:12]: This looks like an old job. It's talking about cypress, missing the job name, etc.

**Review by ermish — CHANGES_REQUESTED** [review]

File: .github/workflows/vitest.yml (hunk 2 above)
Comment (RageZBla) [reply .github/workflows/vitest.yml:12]: @ermish not sure if I am following can you point out the actual syntax or doc you want to use?

Comment (ermish) [reply .github/workflows/vitest.yml:12]: Sorry! I'll add specifics

File: .github/workflows/vitest.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  cypress-run:
```
Comment (ermish) [inline .github/workflows/vitest.yml:13]: This should be related to vitest not cypress
```suggestion
  run-pinia-tests:
```

File: .github/workflows/vitest.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
```
Comment (ermish) [inline .github/workflows/vitest.yml:14]: Add a name section for it to be more user friend in the GitHub PR view

```suggestion
    runs-on: ubuntu-latest
    name: Running Pinia Tests 🚀
```

File: .github/workflows/vitest.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Run ESLint
```
Comment (ermish) [inline .github/workflows/vitest.yml:18]:
```suggestion
      - name: Run Vitest
```

File: .github/workflows/vitest.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        run: yarn run test
```
Comment (ermish) [inline .github/workflows/vitest.yml:19]:
```suggestion
        run: yarn test:pinia
```

**Review by ermish — CHANGES_REQUESTED** [review]
Updated the comments @RageZBla

**Comment by RageZBla** [comment]
@ermish @theyokohamalife finally all :green_circle:

**Review by ermish — APPROVED** [review]
LGTM! Nice job @RageZBla 😎

**Pull request merged**